hardware_interface/include/hardware_interface/actuator.hpp

```diff
@@ -71,10 +71,10 @@ class Actuator final
   const rclcpp_lifecycle::State & error();
 
   HARDWARE_INTERFACE_PUBLIC
-  std::vector<StateInterface> export_state_interfaces();
+  std::vector<std::shared_ptr<StateInterface>> export_state_interfaces();
```

Comment:
```suggestion
  std::vector<std::shared_ptr<const StateInterface>> export_state_interfaces();
```
The StateInterface better use the SharedConstPtr as they are not supposed to be changed outside the hardware interface. After defining the typedef, the following would be better

```suggestion
  std::vector<StateInterface::SharedConstPtr> export_state_interfaces();
```